code/modules/uplink/uplink_items.dm

```diff
@@ -752,6 +753,24 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	surplus = 10
 	exclude_modes = list(/datum/game_mode/nuclear, /datum/game_mode/nuclear/clown_ops, /datum/game_mode/incursion)
 
+/datum/uplink_item/stealthy_weapons/pacifist
+	name = "Paci-Fist Martial Arts Scroll"
+	desc = "This scroll contains the secrets of the ancient martial arts technique of the Paci-Fist. You will learn \
+			various ways to disarm and pacify your foes. Might alter an individuals outlook on life."
+	item = /obj/item/book/granter/martial/pacifist
+	cost = 16
+	surplus = 10
+	exclude_modes = list(/datum/game_mode/nuclear, /datum/game_mode/nuclear/clown_ops) //nukies lose their pacifism trait at roundstart, it would be counterintuitive and cheesy to add this in
+
+/datum/uplink_item/stealthy_weapons/pacifist/restricted
+	category = "Discounted Gear"
+	desc = "This scroll contains the secrets of the ancient martial arts technique of the Paci-Fist. You will learn \
+			various ways to disarm and pacify your foes. Normally costs 16TC, discounted especially for you."
+	cost = 10
+	surplus = 0
+	restricted_roundstart_traits = list(TRAIT_PACIFISM) // Discounted for roundstart trait pacifists, since they already have irrevertible pacifism.
```

Comment:
```suggestion
	restricted_roundstart_traits = list(/datum/quirk/nonviolent) // Discounted for roundstart trait pacifists, since they already have irrevertible pacifism.
```